src/Illuminate/Cookie/CookieJar.php

```diff
@@ -138,18 +146,31 @@ public function queue(...$parameters)
             $cookie = call_user_func_array([$this, 'make'], $parameters);
         }
 
-        $this->queued[$cookie->getName()] = $cookie;
+        if (! isset($this->queued[$cookie->getName()])) {
+            $this->queued[$cookie->getName()] = [];
+        }
+
+        $this->queued[$cookie->getName()][$cookie->getPath()] = $cookie;
     }
 
     /**
      * Remove a cookie from the queue.
      *
      * @param  string  $name
+     * @param  string|null $path
      * @return void
      */
-    public function unqueue($name)
+    public function unqueue($name, $path = null)
     {
-        unset($this->queued[$name]);
+        if ($path === null) {
```

Comment:
I encourage avoiding nested 'ifs' because increasing the difficulty of understanding the code.
See the follow example:
```
if ($path === null) {
    unset($this->queued[$name]);
    return;
}

unset($this->queued[$name][$path]);
if (empty($this->queued[$name])) {
    unset($this->queued[$name]);
}
```